### ISSUE TITLE
Wait X Time Before Looping Attempts or Don't Loop

### ISSUE THREAD

**Issue by cwilliams5** [Luois45/claim-free-steam-packages#10]
**Is your feature request related to a problem? Please describe.**
Currently the script loops on completion. If you are up to data on packages this means it is looping every 30 seconds. That means:
- Scraping the github packages file every 30 seconds
- Always using a full processing core of my machine to endlessly process
- A whole bunch of (unnecessary) I/O disk operations while processing (separate bug/enhancement submitted in #5 and #9)

**Describe the solution you'd like**
Ideally I want to run this script unattended and keep my licenses up to date. One way to obtain this would for the script to sleep in between attempts, for example for 12 hours. The time could even be configurable. That way I could just run this in a docker on the same Raspberry Pi already running ASF and just have my licenses always updated. 

**Describe alternatives you've considered**
Another option would be to NOT loop forever, just cleanly exit. Then a linux cron job or windows scheduled task could be used to run the program on the users desired schedule.

**Additional context**
I hacked this in myself by prefixing the line 108 (return activatedPackage) with:

```
print('Waiting 12 Hours to Check for New Licenses')
time.sleep(43200)
```
Works great, but not ideal to have to redo every time you update.

**Comment by Luois45** [comment]
@cwilliams5
I do think that maybe looping it every hour or every second hour should be better than every 12 hours.
Can you just open a PR with the changes made?

**Comment by cwilliams5** [comment]
> @cwilliams5 I do think that maybe looping it every hour or every second hour should be better than every 12 hours. Can you just open a PR with the changes made?

PR #14 adds a configurable loop delay with progress bar, slight api speed up, and console feedback

**Comment by Luois45** [comment]
I don't want it to be 2 hours because the list updates that often.
The reason is that ASF can't activate a package sometimes while other ones are working.
Thats why I do want it to retry after a certain period.
Thanks for the generous offer. I will try to set up a local docker container or put it onto my Raspberry pi which does update the package list automatically every few hours or so. 
> How long does the package update script take to run?

The package script does take around 1 hour to run with the use of proxies.

**Comment by cwilliams5** [comment]
As a note PR #14 was closed and resubmitted as PR #15 because git broke and I dont have time to troubleshoot.

**Comment by cwilliams5** [comment]
Resolved by #15.
